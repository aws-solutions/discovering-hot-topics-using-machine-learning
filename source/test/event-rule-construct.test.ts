/**********************************************************************************************************************
 *  Copyright 2020-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.                                      *
 *                                                                                                                    *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance    *
 *  with the License. A copy of the License is located at                                                             *
 *                                                                                                                    *
 *      http://www.apache.org/licenses/LICENSE-2.0                                                                    *
 *                                                                                                                    *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES *
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions    *
 *  and limitations under the License.                                                                                *
 *********************************************************************************************************************/

import { buildLambdaFunction } from '@aws-solutions-constructs/core';
import { Stack } from 'aws-cdk-lib';
import { LambdaFunction } from 'aws-cdk-lib/aws-events-targets';
import { Code, Runtime } from 'aws-cdk-lib/aws-lambda';
import { Config, EventRule } from '../lib/integration/event-rule-construct';

test('test Event Rule Construct', () => {
    const stack = new Stack();

    const proxylambda = buildLambdaFunction(stack, {
        lambdaFunctionProps: {
            runtime: Runtime.NODEJS_18_X,
            handler: 'index.handler',
            code: Code.fromAsset(`${__dirname}/../lambda/integration`),
            environment: {
                STREAM_NAME: 'test-stream'
            }
        }
    });

    const configs: Config[] = [
        {
            source: ['com.test'],
            ruleTargets: [new LambdaFunction(proxylambda)]
        }
    ];

    new EventRule(stack, 'EventRule', {
        configs: configs
    });
});
