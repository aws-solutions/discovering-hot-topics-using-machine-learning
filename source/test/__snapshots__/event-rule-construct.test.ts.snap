// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Event Rule Construct 1`] = `
{
  "Parameters": {
    "AssetParametersbe74554f56244971eebbcf26bdd0c0d8d286a63816b49c2feed5c83d4eca2edeArtifactHashCD969C48": {
      "Description": "Artifact hash for asset "be74554f56244971eebbcf26bdd0c0d8d286a63816b49c2feed5c83d4eca2ede"",
      "Type": "String",
    },
    "AssetParametersbe74554f56244971eebbcf26bdd0c0d8d286a63816b49c2feed5c83d4eca2edeS3Bucket1F997E68": {
      "Description": "S3 bucket for asset "be74554f56244971eebbcf26bdd0c0d8d286a63816b49c2feed5c83d4eca2ede"",
      "Type": "String",
    },
    "AssetParametersbe74554f56244971eebbcf26bdd0c0d8d286a63816b49c2feed5c83d4eca2edeS3VersionKey7E521FA2": {
      "Description": "S3 key for asset version "be74554f56244971eebbcf26bdd0c0d8d286a63816b49c2feed5c83d4eca2ede"",
      "Type": "String",
    },
  },
  "Resources": {
    "EventRuleEventRule0AllowEventRuleLambdaFunction73CE9FCC": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "LambdaFunctionBF21E41F",
            "Arn",
          ],
        },
        "Principal": "events.amazonaws.com",
        "SourceArn": {
          "Fn::GetAtt": [
            "EventRuleEventRule0FC2BB61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "EventRuleEventRule0FC2BB61D": {
      "Properties": {
        "EventBusName": {
          "Ref": "EventRuleInferenceEventsB6AAA802",
        },
        "EventPattern": {
          "account": [
            {
              "Ref": "AWS::AccountId",
            },
          ],
          "region": [
            {
              "Ref": "AWS::Region",
            },
          ],
          "source": [
            "com.test",
          ],
        },
        "State": "ENABLED",
        "Targets": [
          {
            "Arn": {
              "Fn::GetAtt": [
                "LambdaFunctionBF21E41F",
                "Arn",
              ],
            },
            "Id": "Target0",
          },
        ],
      },
      "Type": "AWS::Events::Rule",
    },
    "EventRuleInferenceEventsB6AAA802": {
      "Properties": {
        "Name": "Default-AppIntegration",
      },
      "Type": "AWS::Events::EventBus",
    },
    "LambdaFunctionBF21E41F": {
      "DependsOn": [
        "LambdaFunctionServiceRoleDefaultPolicy126C8897",
        "LambdaFunctionServiceRole0C4CDE0B",
      ],
      "Metadata": {
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "W58",
              "reason": "Lambda functions has the required permission to write CloudWatch Logs. It uses custom policy instead of arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole with tighter permissions.",
            },
            {
              "id": "W89",
              "reason": "This is not a rule for the general case, just for specific use cases/industries",
            },
            {
              "id": "W92",
              "reason": "Impossible for us to define the correct concurrency for clients",
            },
          ],
        },
      },
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "AssetParametersbe74554f56244971eebbcf26bdd0c0d8d286a63816b49c2feed5c83d4eca2edeS3Bucket1F997E68",
          },
          "S3Key": {
            "Fn::Join": [
              "",
              [
                {
                  "Fn::Select": [
                    0,
                    {
                      "Fn::Split": [
                        "||",
                        {
                          "Ref": "AssetParametersbe74554f56244971eebbcf26bdd0c0d8d286a63816b49c2feed5c83d4eca2edeS3VersionKey7E521FA2",
                        },
                      ],
                    },
                  ],
                },
                {
                  "Fn::Select": [
                    1,
                    {
                      "Fn::Split": [
                        "||",
                        {
                          "Ref": "AssetParametersbe74554f56244971eebbcf26bdd0c0d8d286a63816b49c2feed5c83d4eca2edeS3VersionKey7E521FA2",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": {
          "Variables": {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
            "STREAM_NAME": "test-stream",
          },
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "LambdaFunctionServiceRole0C4CDE0B",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "TracingConfig": {
          "Mode": "Active",
        },
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaFunctionServiceRole0C4CDE0B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": [
                    "logs:CreateLogGroup",
                    "logs:CreateLogStream",
                    "logs:PutLogEvents",
                  ],
                  "Effect": "Allow",
                  "Resource": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          "Ref": "AWS::Partition",
                        },
                        ":logs:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        {
                          "Ref": "AWS::AccountId",
                        },
                        ":log-group:/aws/lambda/*",
                      ],
                    ],
                  },
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "LambdaFunctionServiceRolePolicy",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LambdaFunctionServiceRoleDefaultPolicy126C8897": {
      "Metadata": {
        "cfn_nag": {
          "rules_to_suppress": [
            {
              "id": "W12",
              "reason": "Lambda needs the following minimum required permissions to send trace data to X-Ray and access ENIs in a VPC.",
            },
          ],
        },
      },
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "xray:PutTraceSegments",
                "xray:PutTelemetryRecords",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "LambdaFunctionServiceRoleDefaultPolicy126C8897",
        "Roles": [
          {
            "Ref": "LambdaFunctionServiceRole0C4CDE0B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
